pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -80,11 +83,33 @@ func (c *cephStatusChecker) checkCephStatus(stopCh chan struct{}) {
 
 // checkStatus queries the status of ceph health then updates the CR status
 func (c *cephStatusChecker) checkStatus() {
+	var status client.CephStatus
+	var err error
+
 	logger.Debugf("checking health of cluster")
-	status, err := client.Status(c.context, c.namespace)
-	if err != nil {
-		logger.Errorf("failed to get ceph status. %v", err)
-		return
+
+	// This is not an external cluster
+	if c.externalCred.Username == "" && c.externalCred.Secret == "" {
```

Comment:
Instead of calling `Status()` three different places, how about just figuring out the user for the different cases, then calling `client.StatusWithUser()` once?